pkg/jira/create.go

```diff
@@ -228,12 +228,22 @@ func constructCustomFields(fields map[string]string, configuredFields []IssueTyp
 	data.Fields.M.customFields = make(customField)
 
 	for key, val := range fields {
+		rawJson := false
+		if strings.HasPrefix(key, "json:") {
+			key = key[5:]
+			rawJson = true
+		}
 		for _, configured := range configuredFields {
```

Comment:
Looks like we still need to go through configured field so why not just add a type called `json` that accepts the raw json data? 

For instance the configuration can be:
```yml
issue:
    fields:
        custom:
            - name: platform
              key: customfield_10032
              schema:
                datatype: json
```

And we can pass it as is

![Screen Shot 2023-09-13 at 9 18 39 PM](https://github.com/ankitpokhrel/jira-cli/assets/2364546/e3fb4ec9-2d69-44c5-863a-36e2211cbf7a)

Now, we don't need to pass `json:` tag?
```sh
jira issue create --custom='platform=[{"value":"Linux"}]'
```

Reply:
It is a good idea to have a `json` type that can be used in the config file. But is there an easy way to update the config file to set the type of custom fields? I think the Jira users on my team will find it hard to locate this file and make the right change by hand. Another inconvenience is that if a user runs `jira init` (for example if their password/access token changes), then the config file will be reset.

Looking at my config file, all the custom fields for which JSON is needed have type `any`. What do you think about using JSON by default for all fields of type `any`?

Another idea is to add an option to the `jira init` command to mark certain fields as needing JSON, e.g.:

```
jira init --json-custom-fields "first-field,second-field,etc"
```

Reply:
> But is there an easy way to update the config file to set the type of custom fields?

Not at the moment unfortunately. There is a plan to have something like [gh config](https://cli.github.com/manual/gh_config) but I don't have any timeline for it yet (open for PRs :)).

> I think the Jira users on my team will find it hard to locate this file and make the right change by hand.

Unless I am missing something, even with the `json:` tag you will still need to configure the field in config right? since we are looping through the configured fields in line#236?

```
for _, configured := range configuredFields {
```

> Looking at my config file, all the custom fields for which JSON is needed have type any. What do you think about using JSON by default for all fields of type any?

> Another idea is to add an option to the jira init command to mark certain fields as needing JSON, e.g.:

These are all valid ideas. The inconsistencies in Jira could make things a bit tricky so I am not sure if this will be the same case for all Jira installations.

I would suggest, let's start with simple and working implementation and we can make it better iteratively.

Reply:
> you will still need to configure the field in config right

Running `jira init` populates the config file with all custom fields, even the ones with unknown (`any`) type, so no manual config is necessary.

The current default behaviour for `any` type fields is to just send a string. I would rather not change that default behaviour in case it breaks something for an existing user (unless you know that this never happens in practice).

So I propose to add `json` as a new custom field type and add a flag to `jira init` to make it easy to configure these fields.

Reply:
Sounds good! Let's start by adding `json` as a new custom field type. We can talk about how we can make the config easily configurable.

Reply:
Here is my second attempt.

- Added `json` as a custom type, which can be used in the config file to force a field to use JSON when setting/editing.
- JSON is used by default for unknown field types, such as `user` or `version`.
- Factored out the code which constructs the `customFieldTypeXxx` objects so that it can be called recursively for arrays of any type, including JSON.
- Replaced the type-specific structs used to edit custom fields on issues, such as `customFieldTypeNumberSet` and `customFieldTypeStringSet` with generic structs that can be used to set/add/remove items on any field type.

I tested this on our Jira installation with custom fields of type `option`, `user`, `version`, `date`, `number`, `string`, `any` and array of `string`. This worked nicely, and I could specify `user` and `version` types as JSON from the command line with no manual changes to the config file needed.

Reply:
@ankitpokhrel Have you had a chance to review my update?

Reply:
Hi @octachrome, sorry for the late response. Would you be able to provide some examples on how to construct valid requests for different types of custom fields? Thanks!

Reply:
@ankitpokhrel, if you mean CLI examples, see below (the syntax is the same for the `issue create` command):

```
# Field of type "version"
jira issue edit --no-input --custom 'found-in-version={"name": "MyProduct 1.0"}' XYZ-1234
# Field of type "user"
jira issue edit --no-input --custom 'assigned-developer={"name": "BobFossil"}' XYZ-1234
```

I could add these to the readme, or it might be better to add them to the article here: https://github.com/ankitpokhrel/jira-cli/discussions/346.